src/modules/endpoint/DatabaseConnector.js

```diff
@@ -72,16 +79,20 @@ type ContentCityJsonType = {|
 type CityCodeType = string
 type LanguageCodeType = string
 
+type MetaCitiesEntryJsonType = {|
+  languages: {
+    [LanguageCodeType]: {|
+      last_update: string
+    |}
+  },
+  last_usage: string | null
```

Comment:
Makes sense, @maxammann what do you think? May we need last usages for languages in the future? Regarding null: Since we are only deleting the resource cache and not categories/events, it makes sense to keep the lastUpdates and set the lastUsage to null. Also, if we are peeking, the `last_update` gets set while the `last_usage` is not since we don't want to keep the resource cache of a just peeking city instead of a normally visited city.

Reply:
Okay one problem now though: If the resource cache is deleted, `shouldRefreshResources` will still be false during the first 24h after the last update of the language. This is in fact also a problem if a city is opened peeking first and afterwards normally. The resource cache won't be fetched then.

Reply:
So I think we either need an additional property which says whether the resource cache exists or not (also in the current language), or we define the `last_usage` property for every language and set it to null if no cache is saved or if it is deleted. Probably a separate property makes more sense (like `lastResourceCachUpdate`) since otherwise the correlation between resource cache and `last_usage`

Reply:
@maxammann @Schedulaar what do you think?

Reply:
> Okay one problem now though: If the resource cache is deleted, shouldRefreshResources will still be false during the first 24h after the last update of the language. This is in fact also a problem if a city is opened peeking first and afterwards normally. The resource cache won't be fetched then.

Wrong: When you navigate via landing or restart the app, we dispatch a `FETCH_CATEGORY` with `shouldRefreshResources: true`.
So the only important thing is that we don't purge the resources of the current city (which we don't).

(`<optionally>`)
Also for the sake of simplicity, I'd suggest to remove not only the resource cache of the least used cities, but all information about them. We could even count in peeking as using the city -- if a city has many dependencies to a different city, we may want to have the peeked city offline as well (at least the json-data).
(`</optionally>`)

Still I don't really get the meaning of `last_usage` to be `null`. What does it tell me? If I see it correctly, it's `null`, if (and only if) the city was once used with resources we deleted or the city was peeked -- basically iff there's no resource cache, right? That's not really helpful, since even if the `lastUsage` is not null, we cannot know, whether the resources are loaded for a particular language (or not) -- resources are language-dependent! Also I don't really expect this information under the cover of `lastUsage`.

What we could do, when deciding which resources (in terms of folders) to delete is: delete all folders in the resources directory, that are not the folders of the top-3-cities.
That way we avoid having to maintain a list of which resource cache has been loaded and which not (which is a highly non-trivial task, since there are most likely always some files which cannot be loaded in the first trial and which should be loaded in a second trial the next time you open the app (or similiar))

Reply:
Yep you're right. I missed `shouldRefreshResources` and thought it's also dependent on `shouldUpdate`. Since that was my reason to argument partly `pro-null`, that changes everything and I agree totally with you now :D I'm not quite sure regarding the peeking thing. @maxammann ?

Reply:
But since including peeking and clearing all the data I think I favor it :man_shrugging: